### PR TITLE
fixed single file blob upload

### PR DESCRIPTION
**Pull request by rajatagrawal-dev**

### CODE REVIEW

File: Tasks/AzureFileCopyV2/Utility.ps1 [hunk 1]
```diff
@@ -247,16 +247,34 @@ function Upload-FilesToAzureContainer
...
+
         $blobPrefix = $blobPrefix.Trim()
         $containerURL = [string]::Format("{0}/{1}/{2}", $blobStorageEndpoint.Trim("/"), $containerName, $blobPrefix).Trim("/")
         $azCopyExeLocation = Join-Path -Path $azCopyLocation -ChildPath "AzCopy.exe"
 
-        Write-Verbose "Executing command: & `"$azCopyExeLocation`" /Source:$sourcePath /Dest:$containerURL /DestKey:`"*****`" $additionalArguments"
+        Write-Verbose "Executing command: & `"$azCopyExeLocation`" /Source:$resolvedSourcePath /Dest:$containerURL /DestKey:`"*****`" $additionalArguments"
 
-        $uploadToBlobCommand = "& `"$azCopyExeLocation`" /Source:$sourcePath /Dest:$containerURL /DestKey:`"$storageKey`" $additionalArguments"
+        $uploadToBlobCommand = "& `"$azCopyExeLocation`" /Source:$resolvedSourcePath /Dest:$containerURL /DestKey:`"$storageKey`" $additionalArguments"
```
Comment (arjgupta) [inline Tasks/AzureFileCopyV2/Utility.ps1]: log the upload to blob command

File: Tasks/AzureFileCopyV2/Utility.ps1 [hunk 2]
```diff
@@ -247,16 +247,34 @@ function Upload-FilesToAzureContainer
...
         Write-Output (Get-VstsLocString -Key "AFC_UploadFilesStorageAccount" -ArgumentList $sourcePath, $storageAccountName, $containerName, $blobPrefix)
 
+        $resolvedSourcePath = $sourcePath
+
+        # Check if source path is a file. If yes, then add /Pattern additional argument.
+        if(Test-Path -Path $sourcePath -PathType Leaf)
+        {
+            $fileInfo = Get-Item $sourcePath
+            $resolvedSourcePath = $fileInfo.Directory.FullName
+            $additionalArguments += " /Pattern:$($fileInfo.Name)"
```
Comment (arjgupta) [inline Tasks/AzureFileCopyV2/Utility.ps1]: fileinfo can have white spaces in name here. It should be enclosed in quotes

Comment (rajatagrawal-dev) [reply Tasks/AzureFileCopyV2/Utility.ps1]: fixed

File: Tasks/AzureFileCopyV2/Utility.ps1 (hunk 1 above)
Comment (rajatagrawal-dev) [reply Tasks/AzureFileCopyV2/Utility.ps1]: already doing that above

File: Tasks/AzureFileCopyV2/Utility.ps1 [hunk 3]
```diff
@@ -275,16 +293,16 @@ function Upload-FilesToAzureContainer
     }
     finally
     {
-        Handle-AzCopyLogs -isLogsPresent $useDefaultArguments -logsFilePath $azCopyLogFilePath -ErrorAction SilentlyContinue
+        Handle-AzCopyLogs -isLogsPresent $useDefaultArguments -logsFilePath "$azCopyLogFilePath" -ErrorAction SilentlyContinue
```
Comment (arjgupta) [inline Tasks/AzureFileCopyV2/Utility.ps1]: nit, you do not need to enclose it in double quotes.

**Pull request merged**